simpleclient/src/main/java/io/prometheus/client/Summary.java

```diff
@@ -38,133 +40,231 @@
  * This would allow you to track request rate, average latency and average request size.
  */
 public class Summary extends SimpleCollector<Summary.Child> {
+    // Quantiles that should be calculated for this Summary
+    // e.g.  0.20, 0.50, 0.99
+    private Quantile[] quantiles;
+    private Stream stream;
+    private boolean shouldCalculateQuant = false;
```

Comment:
I think this field is redundant.
There's no point in using the Summary metric without it having Quantiles.
The quantiles field should contain default values and then the *shouldCalculateQuant* field is always true.
We can have a separate discussion on what the defaults should be (I'd say have a median and some upper quantiles like 0.95, 0.99 and 0.999).